### ISSUE TITLE
Fail to initialize Atlas client, stop this listener

### ISSUE THREAD

**Issue by wgzhao** [hortonworks-spark/spark-atlas-connector#275]
I try to use HDP 3.1 built-in spark and atlas , but some errors occurred.

my env:
HDP 3.1.0.0-78
JDK: Oracle JDK1.8.0_211-amd64 with unlimited JCE policy
OS: CentOS 7.5.1804

I use hive account , and execute the following command:

```shell
spark-sql  --master local --jars /usr/hdp/3.1.0.0-78/spark-atlas-connector/spark-atlas-connector-assembly-0.1.0.3.1.0.0-78.jar --hiveconf  spark.extraListeners=com.hortonworks.spark.atlas.SparkAtlasEventTracker
```
I got the following errors:

```
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
        at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:904)
        at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:198)
        at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:228)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:137)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
19/08/01 16:59:31 INFO SparkContext: Registered listener com.hortonworks.spark.atlas.SparkAtlasEventTracker
19/08/01 16:59:31 INFO SharedState: loading hive config file: file:/etc/spark2/3.1.0.0-78/0/hive-site.xml
19/08/01 16:59:31 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir ('/apps/spark/warehouse').
19/08/01 16:59:31 INFO SharedState: Warehouse path is '/apps/spark/warehouse'.
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@58eac00e{/SQL,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@6acffb2d{/SQL/json,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@1a2e0d57{/SQL/execution,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@5f025000{/SQL/execution/json,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@49c099b{/static/sql,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO HiveUtils: Initializing HiveMetastoreConnection version 3.0 using file:/usr/hdp/current/spark2-client/standalone-metastore/standalone-metastore-1.21.2.3.1.0.0-78-hive3.jar:file:/usr/hdp/current/spark2-client/standalone-metastore/stand
alone-metastore-1.21.2.3.1.0.0-78-hive3.jar
19/08/01 16:59:31 INFO HiveConf: Found configuration file file:/usr/hdp/current/spark2-client/conf/hive-site.xml
Hive Session ID = 1149a1d8-4c31-4019-aa37-c83a02050ce6
19/08/01 16:59:31 INFO SessionState: Hive Session ID = 1149a1d8-4c31-4019-aa37-c83a02050ce6
19/08/01 16:59:31 INFO SessionState: Created HDFS directory: /tmp/spark/admin/1149a1d8-4c31-4019-aa37-c83a02050ce6
19/08/01 16:59:31 INFO SessionState: Created local directory: /tmp/admin/1149a1d8-4c31-4019-aa37-c83a02050ce6
19/08/01 16:59:31 INFO SessionState: Created HDFS directory: /tmp/spark/admin/1149a1d8-4c31-4019-aa37-c83a02050ce6/_tmp_space.db
19/08/01 16:59:31 INFO HiveClientImpl: Warehouse location for Hive client (version 3.0.0) is /apps/spark/warehouse
19/08/01 16:59:31 INFO HiveMetaStoreClient: Trying to connect to metastore with URI thrift://t2.fzzq.com:9083
19/08/01 16:59:31 INFO HiveMetaStoreClient: Opened a connection to metastore, current connections: 1
19/08/01 16:59:31 INFO HiveMetaStoreClient: Connected to metastore.
19/08/01 16:59:31 INFO RetryingMetaStoreClient: RetryingMetaStoreClient proxy=class org.apache.hadoop.hive.ql.metadata.SessionHiveMetaStoreClient ugi=admin (auth:SIMPLE) retries=1 delay=5 lifetime=0
19/08/01 16:59:32 INFO StateStoreCoordinatorRef: Registered StateStoreCoordinator endpoint
spark-sql> 19/08/01 17:05:32 INFO AbstractConnector: Stopped Spark@2620e717{HTTP/1.1,[http/1.1]}{0.0.0.0:4040}
19/08/01 17:05:32 INFO SparkUI: Stopped Spark web UI at http://t1.fzzq.com:4040
19/08/01 17:05:32 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
        at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:904)
        at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:198)
        at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:228)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:137)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
19/08/01 16:59:31 INFO SparkContext: Registered listener com.hortonworks.spark.atlas.SparkAtlasEventTracker
19/08/01 16:59:31 INFO SharedState: loading hive config file: file:/etc/spark2/3.1.0.0-78/0/hive-site.xml
19/08/01 16:59:31 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir ('/apps/spark/warehouse').
19/08/01 16:59:31 INFO SharedState: Warehouse path is '/apps/spark/warehouse'.
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@58eac00e{/SQL,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@6acffb2d{/SQL/json,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@1a2e0d57{/SQL/execution,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@5f025000{/SQL/execution/json,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO ContextHandler: Started o.s.j.s.ServletContextHandler@49c099b{/static/sql,null,AVAILABLE,@Spark}
19/08/01 16:59:31 INFO HiveUtils: Initializing HiveMetastoreConnection version 3.0 using file:/usr/hdp/current/spark2-client/standalone-metastore/standalone-metastore-1.21.2.3.1.0.0-78-hive3.jar:file:/usr/hdp/current/spark2-client/standalone-metastore/stand
alone-metastore-1.21.2.3.1.0.0-78-hive3.jar
19/08/01 16:59:31 INFO HiveConf: Found configuration file file:/usr/hdp/current/spark2-client/conf/hive-site.xml
....
```

To verify my configuration , I used the following command:

```
curl -u admin:<mypassword> http://localhost:21000/api/atlas/admin/version
{"Description":"Metadata Management and Data Governance Platform over Hadoop","Revision":"6f92b92c82911d58d8f49cd46d2c2a6d89196b9c","Version":"1.1.0.3.1.0.0-78","Name":"apache-atlas"}
```
It seems like ok.

I try to compile Atlas 2.0.0/1.0.0 and spark-atlas-connector with spark 2.4.3 version ,but all attempts are failed.

what's missing in my configuration ?
or how can i specify username and password connected to Atlas in command line .

I have take almost 2 weeks to try to fix it , but no good message brings to me.

**Comment by wgzhao** [comment]
I have resolved the problem. 
I add the following two lines to `/etc/spark2/conf/atlas-application.properties`

```
atlas.client.username=admin
atlas.client.password=<mypassword>
```

**Comment by aemanukyan** [comment]
@wgzhao I'm working with HDP 3.1 too and I want to make a simple example so I setup everything for apache 1.1 and spark 2.3 because it's included into HDP 3.1 and copied zeppelin script. When I start a zeppeling script nothing appears in Atlas but job is successfully completed.
